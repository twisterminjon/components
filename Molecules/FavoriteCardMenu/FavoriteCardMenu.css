/* Note: See UserCardFavorite.css for how the visibility of the menu is controlled.
We are using the hover of the fav button to control the favoritecardmenu width. */
.favoritecardmenu {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;

  top: 0;
  left: 0;

  width: 0;
  height: 76px;

  background-color: var(--favcard__menu_bg);
  border: none;
  border-radius: 4px;
  outline: none;

  overflow: hidden;
  z-index: 1;
}

.favoritecardmenu-button {
  display: flex;
  align-items: center;

  width: 100%;
  max-width: 100px;
  height: 30px;

  padding: 4px 6px;
  margin: 6px 8px;

  background-color: var(--button_color);

  border: none;
  border-radius: 4px;
  outline: none;
  text-align: left;
}

.favoritecardmenu-icon {
  height: 16px;
  margin-right: 6px;
  flex-shrink: 0;
}

.favoritecardmenu-button-text {
  font-size: 16px;
  font-weight: 700;
}

.favoritecardmenu-button:hover {
  background-color: var(--button_color_hover);
}

.favoritecardmenu-button:focus {
  background-color: var(--button_color_hover);
}

.favoritecardmenu-button:active {
  background-color: var(--button_color_pressed);
}

.favoritecardmenu-button:disabled,
.favoritecardmenu-button[disabled] {
  background-color: var(--favcard__menu__button_bg);
  color: var(--favcard__menu__button_fg);
  pointer-events: none;
}
