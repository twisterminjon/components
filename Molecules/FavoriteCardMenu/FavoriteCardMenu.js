import React, { Component } from 'react'
import PropTypes from 'prop-types'

import IconMessage from '../../Atoms/Icons/IconMessage'
import IconAnswerCall from '../../Atoms/Icons/IconAnswerCall'

import './FavoriteCardMenu.css'

export default class FavoriteCardMenu extends Component {
  static propTypes = {
    /** If true, calling will be disabled */
    disableCall: PropTypes.bool.isRequired,
    /** If true, messaging will be disabled */
    disableMessages: PropTypes.bool,
    /** When call button is clicked */
    onCall: PropTypes.func.isRequired,
    /** When the message button is clicked */
    onMessage: PropTypes.func.isRequired,
    /** A label for the testid on the buttons */
    testidLabel: PropTypes.string,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    testidLabel: '',
    className: '',
    style: {},
  }

  render() {
    const { disableCall, disableMessages, onCall, onMessage, testidLabel, className, style } = this.props

    // FIXME: icons should use the css color to get their color instead of a prop
    const callButtonColor = disableCall ? 'rgba(0, 0, 0, 0.2)' : 'var(--black)'

    return (
      <div className={`favoritecardmenu ${className}`} data-testid={`fav-menu-${testidLabel}`} style={style}>
        <button
          className="favoritecardmenu-button"
          onClick={onCall}
          disabled={disableCall}
          data-testid={`fav-call-${testidLabel}`}>
          <IconAnswerCall className="favoritecardmenu-icon" color={callButtonColor} style={{ width: 16 }} size={16} />
          <span className="favoritecardmenu-button-text">Call</span>
        </button>
        <button
          className="favoritecardmenu-button"
          onClick={onMessage}
          disabled={disableMessages}
          data-testid={`fav-message-${testidLabel}`}>
          <IconMessage className="favoritecardmenu-icon" size={16} />
          <span className="favoritecardmenu-button-text">Message</span>
        </button>
      </div>
    )
  }
}
