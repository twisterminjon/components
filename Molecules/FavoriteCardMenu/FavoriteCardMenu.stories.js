import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import FavoriteCardMenu from './FavoriteCardMenu'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('FavoriteCardMenu', () => {
  const disableCallLab = 'disableCall'
  const disableCallDef = false
  const disableCallVal = boolean(disableCallLab, disableCallDef)

  return (
    <div
      style={{
        backgroundImage: 'url("https://picsum.photos/500/500/?random")',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      }}>
      <FavoriteCardMenu
        // Note: style is just so it shows in the story (it is width: 0 by default and width is controlled by css in another component)
        style={{ width: 116 }}
        disableCall={disableCallVal}
        onCall={() => {
          alert('Call clicked')
        }}
        onMessage={() => {
          alert('Message clicked')
        }}
      />
    </div>
  )
})
