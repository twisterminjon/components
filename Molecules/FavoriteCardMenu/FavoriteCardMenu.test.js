import React from 'react'
import FavoriteCardMenu from './FavoriteCardMenu'

describe('FavoriteCardMenu', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <FavoriteCardMenu show={true} disableCall={false} onCall={mockFun} onMessage={mockFun} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when calling is disabled', () => {
    const wrapper = window.shallow(
      <FavoriteCardMenu show={true} disableCall={true} onCall={mockFun} onMessage={mockFun} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
