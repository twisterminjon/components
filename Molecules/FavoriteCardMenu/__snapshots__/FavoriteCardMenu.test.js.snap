// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCardMenu matches the snapshot 1`] = `
<div
  className="favoritecardmenu "
  data-testid="fav-menu-"
  style={Object {}}
>
  <button
    className="favoritecardmenu-button"
    data-testid="fav-call-"
    disabled={false}
    onClick={[MockFunction]}
  >
    <IconAnswerCall
      className="favoritecardmenu-icon"
      color="var(--black)"
      size={16}
      style={
        Object {
          "width": 16,
        }
      }
    />
    <span
      className="favoritecardmenu-button-text"
    >
      Call
    </span>
  </button>
  <button
    className="favoritecardmenu-button"
    data-testid="fav-message-"
    onClick={[MockFunction]}
  >
    <IconMessage
      className="favoritecardmenu-icon"
      color="black"
      size={16}
    />
    <span
      className="favoritecardmenu-button-text"
    >
      Message
    </span>
  </button>
</div>
`;

exports[`FavoriteCardMenu matches the snapshot when calling is disabled 1`] = `
<div
  className="favoritecardmenu "
  data-testid="fav-menu-"
  style={Object {}}
>
  <button
    className="favoritecardmenu-button"
    data-testid="fav-call-"
    disabled={true}
    onClick={[MockFunction]}
  >
    <IconAnswerCall
      className="favoritecardmenu-icon"
      color="rgba(0, 0, 0, 0.2)"
      size={16}
      style={
        Object {
          "width": 16,
        }
      }
    />
    <span
      className="favoritecardmenu-button-text"
    >
      Call
    </span>
  </button>
  <button
    className="favoritecardmenu-button"
    data-testid="fav-message-"
    onClick={[MockFunction]}
  >
    <IconMessage
      className="favoritecardmenu-icon"
      color="black"
      size={16}
    />
    <span
      className="favoritecardmenu-button-text"
    >
      Message
    </span>
  </button>
</div>
`;
